### ISSUE TITLE
Non-concrete fields creates malformed SQL on `upsert`

### ISSUE THREAD

**Issue by dalberto** [Opus10/django-pgbulk#29]
First, thanks for this library and all the other django<>postgres libraries, it's super clean and useful. I love and am using `pglock` as well!

I've tried to use `pgbulk.upsert` on a model with a virtual/proxy field. Specifically, I'm using [hashids](https://github.com/ericls/django-hashids) with a field declaration like the following:

```
public_id = HashidsField(real_field_name="id", min_length=6)
```

When calling `upsert` with this field present, I get an error like the following:

```
django.db.utils.ProgrammingError: syntax error at or near "None"
LINE 1: ...:numeric(5, 2), 'some string'::text, NULL::None), ('2...
```

It appears the translation from field definition to SQL is incorrect, indicated by the `NULL::None` in that error message.

Removing the virtual field resolves the issue. Excluding the field via `upsert`'s  `exclude` has no effect unfortunately.

**Comment by dalberto** [comment]
Fix was straightforward, addressed in #30

**Comment by awmath** [comment]
This also applies to the new GeneratedField in django > 5.0 for postgres. In that case the field has the attribute `concrete = True` but  also `generated = True`.
I will also create a PR for that case. We'll see how this goes.

**Comment by wesleykendall** [comment]
Thanks for the PR for the fix @dalberto ! I will go ahead and ensure that generated fields are tested as well @awmath. Will close this issue and report back when deployed.

**Comment by wesleykendall** [comment]
Fixed in v2.4. Generated fields are supported too
